src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -463,6 +518,12 @@ namespace syrec {
         // while there are nodes left to process.
         while (!queue.empty()) {
             const auto current = queue.front();
+
+            //The synthesis is stopped after the nodeThreshold is reached (considered only while synthesizing the decoder)
+            if (current.p->v == nodeThreshold - 1 && flag) {
```

Comment:
I'd rather exchange both of the conditions, i.e., 
```suggestion
            if (flag && (current.p->v == nodeThreshold - 1)) {
```